openapi/src/parts/azure_landing_zone.yaml

```diff
@@ -0,0 +1,196 @@
+# Azure Landing Zones paths
+paths:
+  /api/azure/landingzones/v1:
+    post:
+      summary: Create Azure landing zone
+      operationId: createAzureLandingZone
+      tags: [ LandingZones ]
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateAzureLandingZoneRequestBody'
+      responses:
+        '200':
+          $ref: '#/components/responses/CreateAzureLandingZoneResponse'
+        '202':
+          $ref: '#/components/responses/CreateAzureLandingZoneResponse'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '403':
+          $ref: '#/components/responses/PermissionDenied'
+        '500':
+          $ref: '#/components/responses/ServerError'
+
+  /api/azure/landingzones/v1/create-result/{jobId}:
+    parameters:
+      - $ref: '#/components/parameters/JobId'
+    get:
+      summary: Get the result of a async job to create an Azure Landing Zone
+      operationId: getCreateAzureLandingZoneResult
+      tags: [ LandingZones ]
+      responses:
+        '200':
+          $ref: '#/components/responses/CreateAzureLandingZoneResponse'
+        '202':
+          $ref: '#/components/responses/CreateAzureLandingZoneResponse'
+        '400':
+          $ref: '#/components/responses/BadRequest'
+        '403':
+          $ref: '#/components/responses/PermissionDenied'
+        '500':
+          $ref: '#/components/responses/ServerError'
+
+  /api/azure/landingzones/definitions/v1:

```

Comment:
This API doesn't return the created landing zones, but instead it returns the list of available definitions (or templates) for landing zone creation. The use case is following: a client invokes this endpoint to get list of different landing zone definitions. Each of these definitions provide different landing zone configurations. Then it uses the name of the definition and version of the definition to create landing zone with specific configuration.

Reply:
I would prefer calling this `templates` both here and in the create structure.

Reply:
I'd avoided the term `templates` because, in the Azure world, it typically refers to ARM templates, which we are not using in this implementation. 

The internal implementation has a concept of a definition, which is where you define the resources in a landing zone that the service must deploy. So, I am okay with surfacing this concept for the sake of consistency.